toolbox/commands/copy_dictionary.py

```diff
@@ -0,0 +1,100 @@
+from typing import Optional
+
+from toolbox.api.datagalaxy_api import get_access_token, Token
+from toolbox.api.datagalaxy_api_dictionary import DataGalaxyApiDictionary
+from toolbox.api.datagalaxy_api_workspaces import DataGalaxyApiWorkspace, DataGalaxyBulkResult
+
+
+def copy_dictionary(url_source: str,
+                    url_target: Optional[str],
+                    token_source: str,
+                    token_target: Optional[str],
+                    workspace_source_name: str,
+                    workspace_target_name: str) -> DataGalaxyBulkResult:
+    if token_target is None:
+        token_target = token_source
+
+    if url_target is None:
+        url_target = url_source
+
+    integration_token_source = Token(token_source)
+    integration_token_target = Token(token_target)
+    source_access_token = get_access_token(url_source, integration_token_source)
+    target_access_token = get_access_token(url_target, integration_token_target)
+    workspaces_api_on_source_env = DataGalaxyApiWorkspace(
+        url=url_source,
+        access_token=source_access_token)
+
+    workspaces_api_on_target_env = DataGalaxyApiWorkspace(
+        url=url_target,
+        access_token=target_access_token
+    )
+
+    dictionary_on_source_workspace = DataGalaxyApiDictionary(
+        url=url_source,
+        access_token=source_access_token,
+        workspace=workspaces_api_on_source_env.get_workspace(workspace_source_name)
+    )
+    dictionary_on_target_workspace = DataGalaxyApiDictionary(
+        url=url_target,
+        access_token=target_access_token,
+        workspace=workspaces_api_on_target_env.get_workspace(workspace_target_name)
+    )
+
+    if workspaces_api_on_target_env.get_workspace(workspace_target_name):
```

Comment:
we are calling `workspaces_api_on_target_env.get_workspace(workspace_target_name)` twice, we should save the result in a variable:

```py
target_workspace = workspaces_api_on_target_env.get_workspace(workspace_target_name)
```

Reply:
We should also make sure that both the source and target workspace exist before continuing:

```py
if target_workspace is None or source_workspace is None:
    raise Exception(f'workspace {workspace_target_name} does not exist')
```